gdal/frmts/prf/phprfdataset.cpp

```diff
@@ -0,0 +1,518 @@
+    }
+
+    int GetOverviewCount()
+    {
+        if( osOverview.size() > 0 )
+        {
+            return (int)osOverview.size();
+        }
+        else
+        {
+            return VRTSourcedRasterBand::GetOverviewCount();
+        }
+    }
+
+    GDALRasterBand* GetOverview( int i )
+    {
+        size_t	n = (size_t)i;
+        if( n < osOverview.size() )
+        {
+            return osOverview[ n ];
+        }
+        else
+        {
+            return VRTSourcedRasterBand::GetOverview( i );
+        }
+    }
+};
+
+class PhPrfDataset : public VRTDataset
+{
+public:
+    PhPrfDataset( GDALAccess eAccess, int nSizeX, int nSizeY, int nBandCount, GDALDataType eType, const char* pszName );
+    ~PhPrfDataset();
+    int AddTile( const char* pszPartName, GDALAccess eAccess, int nWidth, int nHeight, int nOffsetX, int nOffsetY, int nScale );
+    static int Identify( GDALOpenInfo* poOpenInfo );
+    static GDALDataset* Open( GDALOpenInfo* poOpenInfo );
+};
+
+PhPrfDataset::PhPrfDataset( GDALAccess _eAccess, int nSizeX, int nSizeY, int nBandCount, GDALDataType eType, const char* pszName ) : VRTDataset( nSizeX, nSizeY )
+{
+    poDriver = (GDALDriver*)GDALGetDriverByName( PH_PRF_DRIVER );
+    eAccess = _eAccess;
+    SetWritable( FALSE );   //Avoid rewrite of *.prf file with 'vrt' file
+    SetDescription( pszName );
+
+    for( int i = 0; i != nBandCount; ++i )
+    {
+        PhPrfBand*	poBand = new PhPrfBand( this, i + 1, eType, nSizeX, nSizeY );
+        SetBand( i + 1, poBand );
+    }
+}
+
+PhPrfDataset::~PhPrfDataset()
+{
+}
+
+int PhPrfDataset::AddTile( const char* pszPartName, GDALAccess eAccessType, int nWidth, int nHeight, int nOffsetX, int nOffsetY, int nScale )
+{
+    GDALProxyPoolDataset*   poTileDataset;
+    poTileDataset = new GDALProxyPoolDataset( pszPartName, nWidth, nHeight,
+                                              eAccessType, FALSE );
+
+    if( poTileDataset == NULL )
+    {
+        return -1;
+    }
+
+    for( int nBand = 1; nBand != GetRasterCount() + 1; ++nBand )
+    {
+        PhPrfBand*      poBand = dynamic_cast<PhPrfBand*>( GetRasterBand( nBand ) );
+        GDALRasterBand* poTileBand;
+
+        if( poBand == NULL )
+        {
+            delete poTileDataset;
+            return -1;
+        }
+
+        //! \todo What reason for nBlockXSize&nBlockYSize passed to AddSrcBandDescription
+        poTileDataset->AddSrcBandDescription( poBand->GetRasterDataType(), 0, 0 );
+        poTileBand = poTileDataset->GetRasterBand( nBand );
+
+        if( 0 == nScale )
+        {
+            poBand->AddSimpleSource( poTileBand, 0, 0, nWidth, nHeight,
+                                     nOffsetX, nOffsetY, nWidth, nHeight );
+        }
+        else
+        {
+            poBand->AddOverview( poTileBand );
+        }
+    }
+    return 0;
+}
+
+int PhPrfDataset::Identify( GDALOpenInfo* poOpenInfo )
+{
+    if( EQUAL( CPLGetExtension( poOpenInfo->pszFilename ), PH_PRF_EXT ) )
```

Comment:
Identification by extension only is fragile. As I see later that there must be a 'phini' root node, perhaps checking for this string in poOpenInfo->pabyHeader (test that it is not NULL) would make identification more robust